### DISCUSSION TITLE
GUI Concept for YT-Spammer-Purge

### DISCUSSION THREAD

**Discussion by juneja108** [ThioJoe/YT-Spammer-Purge#855] [Ideas]
![Login Screen - YT Spammer Purge](https://user-images.githubusercontent.com/74201658/173130612-777752a7-e7cf-474f-b5cd-96a1679be97b.png)

So, I made a GUI concept for this project.. I think the next step is to make all the pages. Then after that I might make a functioning GUI in Python(it won't interact with the YouTube API, unless someone makes an API of this).

I'm pretty sure I can use QT to make the GUI in Python..

If you have any suggestions for this mockup, please let me know by commenting on this post! If you guys also want the Figma file for this, then let me know!

**Comment by Firecul** [comment]
I like the look of this one, there is a large discussion of different ideas here https://github.com/ThioJoe/YT-Spammer-Purge/discussions/84

**Reply by juneja108** [reply]
Thanks :)

**Comment by juneja108** [comment]
## **UPDATE**
Just discovered a library for Python that will speed up the process of the making of this GUI. It is called Eel. It allows you to use HTML, CSS, and Javascript for the frontend(allowing me to make the UI I designed), and Python for the backend. 

If TheoJoe is seeing this post, you now don't need to have to re-write the program in C# just to make the UI.

## **Links**
[Eel Pypi Page](https://pypi.org/project/Eel/)
[Eel Official Discord Server](https://discord.com/invite/3nqXPFX)

**Comment by juneja108** [comment]
## **I NEED HELP**
I created a lot of prototypes for the desktop app, but I need help with dashboards for triggering the mass purge/auto report.
Does anyone have an idea on what I can do?

Comment on this post with your ideas. If you would like a community version of the Figma file, then please let me know!

## **Links**
[Figma File (Read Only)](https://www.figma.com/file/HedWLXFFOsok6Y51KZ9G69/YT-Spammer-Purge-GUI?node-id=0%3A1)

**Reply by Firecul** [reply]
If you are using Figma, have you made a FigJam file to figure out what the UI will do/in what way, ie how a user will interact and get to each option?
What features do you intend to support?

**Reply by juneja108** [reply]
> If you are using Figma, have you made a FigJam file to figure out what the UI will do/in what way, ie how a user will interact and get to each option? What features do you intend to support? - **Firecul**

1. Hmm.. good idea with the FigJam file idea(although I haven't actually used FigJam before, so I'm going to have to look into it)
2. I think I'm going to try to tackle as many features as I can, but I might not make the app fully functional(just the window), because the code using AI to search for spam comments is a secret, and ThioJoe has not put it in the repository to my knowledge. What I plan on to do is either work with TheoJoe to make the app functional, or make a public repo with the barebones of the desktop app(so people can fork it and add their own functions) .

**Comment by TechStudent10** [comment]
Nice concept!

**Reply by juneja108** [reply]
Thanks!

**Comment by SilverscaleClockwork** [comment]
Implementing this will take a while, since the command line version isn't modular enough. (yet)

**Reply by juneja108** [reply]
Yeah, it would take a while to do. I think I might only make the Figma concepts..
